packages/typegpu/src/data/numeric.ts

```diff
@@ -4,15 +4,16 @@ import { inGPUMode } from '../gpuMode';
 import type { TgpuData } from '../types';
 import { SimpleTgpuData } from './std140';
 
-const primitiveNumeric = (
+const primitiveNumeric = <TKind extends string>(
   schema: TB.Uint32Schema | TB.Float32Schema | TB.Int32Schema,
-  code: string,
+  code: TKind,
 ) => {
   return {
     // Type-token, not available at runtime
     __unwrapped: undefined as unknown as number,
-    isLoose: false as const,
 
+    isLoose: false as const,
+    kind: code,
     size: 4,
     byteAlignment: 4,
     expressionCode: code,
```

Comment:
Is it necessary to have both `kind` and `expressionCode`?

Reply:
I wanted to eventually move away from having `expressionCode` as a public member, since code should be retrieved through `resolve`ing. I do not want people to depend on this property being available, and `kind` would not have to be a 1-to-1 WGSL representation, but instead just a unique string that identifies that schema.